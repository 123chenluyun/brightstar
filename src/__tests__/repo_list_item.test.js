import React from 'react';
import ReactDOM from 'react-dom';
import RepoListItem from '../components/repo_list_item';
import { mount } from 'enzyme';
import sinon from 'sinon';
import renderer from 'react-test-renderer';

describe('<RepoListItem />', () => {

	let repo_list_item, props;

	beforeEach(() => {
	  props = {
	    repo: {id:1, contributors_url:"www.test.com",  stargazers_count: 100, name: "test", description: "test description" }
	  };
	  repo_list_item = mount(<table><tbody><RepoListItem {...props}/></tbody></table>); 
	}); 

	it('renders correctly', () => {
	  const repo_list_item = renderer.create(
	   <table><tbody><RepoListItem {...props}/></tbody></table>
	  ).toJSON();
	  expect(repo_list_item).toMatchSnapshot();
	});

	it('calls componentWillMount', () => {
	   sinon.spy(RepoListItem.prototype, 'componentWillMount');
	   repo_list_item = mount(<table><tbody><RepoListItem {...props}/></tbody></table>); 
	   expect(RepoListItem.prototype.componentWillMount.calledOnce).toEqual(true);
	 });

	it('RepoListItem has proper elements', () => {
	  expect(repo_list_item.find('tr').length).toEqual(1);
	  expect(repo_list_item.find('td').length).toEqual(3);
	  expect(repo_list_item.find('h4').length).toEqual(1);
	});

});