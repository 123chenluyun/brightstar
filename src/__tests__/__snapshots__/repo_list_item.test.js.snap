// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RepoListItem /> renders correctly 1`] = `
<table>
  <tbody>
    <tr>
      <td
        className="star-cell"
      >
        <svg
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width="1em"
        >
          <g>
            <path
              d="m37.5 15l-12.2-1.6-5.3-10.9-5.3 10.9-12.2 1.6 9 8.2-2.3 11.8 10.8-5.8 10.8 5.8-2.3-11.8 9-8.2z"
            />
          </g>
        </svg>
        100
      </td>
      <td
        className="repo-details-cell"
      >
        <h4>
          test
        </h4>
        test description
      </td>
      <td
        className="contributors-cell"
      />
    </tr>
  </tbody>
</table>
`;
